src/client/components/SortComponent/Sorting.component.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './Sorting.styles.css';
+
+export default function SortComponent({ products, setSortedProducts }) {
```

Comment:
Overall, I find it very strange that the SortComponent actually does the sorting itself. Further, it can only do sorting of Products. 

I think we're missing the goal of making components re-usable and considering what logic should live where in your app.

I would expect the app to work more like this:
```js
const productSortOptions = [
  'AlphabeticallyAZ',
  'AlphabeticallyZA',
  'PriceDescending',
  'PriceAscending',
];

function App() {
  const [sortState, setSortState] = useState(productSortOptions[0]); // It makes sense to apply a sorting by default
  const unsortedProducts = useProducts(); // Might return the data unsorted from the API
  const [products, setProducts] = useState(unsortedProducts);

  useEffect(() => {
    setProducts(sortProducts(sortState, [...unsortedProducts]));
  }, [unsortedProducts, sortState]);

  return (
    <div>
      <SortComponent sortState={sortState} onSortStateChange={setSortState} sortOptions={productSortOptions} />
      <ProductList products={products} />
    </div>
  );
}
```

This does a few things. 
a) `ProductList` knows noting about where the products come from, if they are sorted or not, it just knows to show a list of products
b) SortComponent knows nothing about products, it could be used to sort anything.
c) App knows nothing about the detals of how to render a sorting selector or a product detail card. It just knows about the state, how that state should change and which component needs which data.